packages/reusables/src/components/ui/input-otp.tsx

```diff
@@ -0,0 +1,153 @@
+import * as React from 'react';
+import { Pressable, Text, View, type ViewProps } from 'react-native';
+import Animated, { withRepeat, withSequence, withTiming, useAnimatedStyle } from 'react-native-reanimated';
+import { cn } from '../../lib/utils';
+import { Input } from './input';
+
+interface InputOTPProps extends ViewProps {
+  value?: string;
+  maxLength?: number;
+  onValueChange?: (value: string) => void;
+  pattern?: RegExp;
+  separator?: React.ReactNode;
+  containerClassName?: string;
+  onBlur?: () => void;
+  groupSize?: number; // Size of each group (e.g., 2 for XX-XX-XX format)
+}
```

Comment:
There's a pretty big difference between the input-otp used by shadcn/ui https://github.com/guilhermerodz/input-otp/blob/master/packages/input-otp/src/types.ts.